### ISSUE TITLE
Legend placement goes haywire when any of the aspect ratio classes are used

### ISSUE THREAD

**Issue by ghost** [CodeYellowBV/chartist-plugin-legend#28]
I am using the "ct-major-second" class for making my chart responsive and auto resizing the bars and stuff.... but any of these classes set the css "top" property to "0" hence making the legend overlap. This can be fixed using custom css, but it isnt recommended as it wont keep the chart and the legend dynamic (i have many pages that use these charts, few having 2 bars, few 5, few 10, etc etc) and i wont be able to handle the height of the legend in all cases if i keep it static.

Is there any way in which i could place the legend outside the entire chart container <div class="ct-chart ct-major-second"></div> as it would be very easy to handle such a situation in that case.

If not how do you suggest i could handle this...?

**Comment by devzing** [comment]
Affecting me as well. Using ct-major-tenth

**Comment by tarekis** [comment]
Well, chartist-plugin-legend places the `ul` element inside of the chart container, either before or after the `svg` chart, [depending on your options](https://github.com/CodeYellowBV/chartist-plugin-legend/blob/master/chartist-plugin-legend.js#L110).

If you want the `ul` to be placed, this has to be done with JS, probably the best approach ist to fork and implement a position option for flexibilty:

```
chart.on('created', function (data) {
    // Append the legend element to the DOM
    switch (options.position) {
        case 'top':
            chart.container.insertBefore(legendElement, chart.container.childNodes[0]);
            break;

        case 'bottom':
            chart.container.insertBefore(legendElement, null);
            break;

        case 'outertop':
            // Some logic to place the chart outside of the chart container
            break;
        }
});
```

How good that is and if that should be PR'd into the master sometimes, i can't say really; you might run into some unexpected styling problems because the legend is now somewhere outside the container, depending on your implementation.

**Comment by SpaceK33z** [comment]
We could also allow the `position` option to take a DOM element. If a DOM element is given it attaches to that element instead of inside the chart. A PR is welcome for this, don't have the time myself atm.

**Comment by tarekis** [comment]
That sounds sweet too, might look into it sometime soon.

**Comment by tarekis** [comment]
gh-pages implementation coming soon too, just messed something up, gonna fix that if that PR makes it.

**Comment by tarekis** [comment]
Works as of `0.6.0`. You can now enter any `HTMLElement` into the position option.

**Comment by tarekis** [comment]
I'm closing this as this should be resolved as of `0.6.0`. If this issue remains, feel free to reopen it.
